kube-core/src/schema.rs

```diff
@@ -39,9 +40,20 @@ impl Visitor for StructuralSchemaRewriter {
                 if let Schema::Object(SchemaObject {
                     instance_type: variant_type,
                     object: Some(variant_obj),
+                    metadata: variant_metadata,
                     ..
                 }) = variant
                 {
+                    if let Some(variant_metadata) = variant_metadata {
+                        // Move enum variant description from oneOf clause to its corresponding property
+                        if let Some(description) = std::mem::take(&mut variant_metadata.description) {
+                            if let Some(Schema::Object(variant_object)) = only_item(variant_obj.properties.values_mut()) {
+                                let metadata = variant_object.metadata.get_or_insert_with(|| Box::new(Metadata::default()));
+                                metadata.description = Some(description.to_string());
```

Comment:
Shouldn't `description` here already be a `String`?

Reply:
Good catch, indeed it is. In a previous version it wasn't.